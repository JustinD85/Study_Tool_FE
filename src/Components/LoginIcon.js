import React, { Component } from 'react';

class LoginIcon extends Component {

  render() {
    const { firstName, email } = this.props.person;

    return (
      <button className="login-profiles" onClick={() => this.props.login(email)}>
        {firstName}
      </button>
    );
  }
}

export default LoginIcon;
