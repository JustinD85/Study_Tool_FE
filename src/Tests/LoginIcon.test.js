import React from 'react';
import LoginIcon from '../Components/LoginIcon';
import { shallow } from 'enzyme';

describe('<LoginIcon/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <LoginIcon person={{firstName:'J', email:'e'}} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});