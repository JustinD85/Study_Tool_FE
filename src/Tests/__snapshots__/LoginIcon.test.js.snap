// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginIcon/> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginIcon
    person={
      Object {
        "email": "e",
        "firstName": "J",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "J",
      "className": "login-profiles",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "J",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "J",
        "className": "login-profiles",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "J",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
